### PR TITLE
Add docs

### PR DESCRIPTION
**Pull request by democat3457**
Adds `typedoc` to provide auto-generated documentation in a webpage format (similar to JavaDocs for Java). Generate new docs by running `npx typedoc` (also mentioned in the readme).

### CODE REVIEW

**Comment by democat3457** [comment]
I could also add a GitHub Action to run the documentation automatically on push to the main branch.

File: src/server/robot/robot.ts
[the remark this answers is not in the document]
Comment (AlexKempen) [reply src/server/robot/robot.ts:1]: Bonus points for adding it to the eslint configuration (You'll have to manually enable `require-hyphen-before-param-description`:
https://www.npmjs.com/package/eslint-plugin-jsdoc

**Review by AlexKempen — COMMENTED** [review]
We now have both a `docs` and a `documentation` folder, which isn't my favorite thing. We also have the github wiki and the external website? I'm also not sure how to actually view the documentation - can we get it to show up in the wiki?

**Comment by AlexKempen** [comment]
I think we'll need an action to push docs to the wiki:
https://github.com/marketplace/actions/publish-to-github-wiki
We'll also need to get .md files, which means something like this:
https://github.com/thlorenz/jsdoc-githubify
Or this:
https://www.npmjs.com/package/jsdoc-to-markdown
Setting up the action would also let us move all of our other documentation to the main repo, which also seems like a good thing.

**Review by xXDMOGXx — APPROVED** [review]
You said it works, so I'll trust ya

**Pull request merged**